tools/eslint-rules/buffer-constructor.js

```diff
@@ -0,0 +1,28 @@
+/**
+ * @fileoverview Require use of new Buffer constructor methods in lib
+ * @author James M Snell
+ */
+'use strict';
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+const msg = 'Use of the Buffer() constructor has been deprecated. ' +
+            'Please use either Buffer.alloc(), Buffer.allocUnsafe(), ' +
+            'or Buffer.from()';
+
+function test(context, node, which) {
+  if (node.type === which && node.callee.name === 'Buffer') {
+    const fn = context.getFilename(node);
```

Comment:
Don't compare file names in the rule. Instead, use the relevant `.eslintrc` to enable or disable the rule.